File: JULY-2025/17-06-1015/p3202.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maximumLength = function(nums, k) {
    const dp = Array.from({length: nums.length}).map(() => Array.from({length: k}).fill(1));
    let max = 0;
    for (let i = nums.length - 1; i >= 0; i --) {
        for (let j = i + 1; j < nums.length; j ++) {
            const remainder = (nums[i] + nums[j]) % k;
            dp[i][remainder] = Math.max(dp[j][remainder] + 1, dp[i][remainder]);
            max = Math.max(max, dp[i][remainder]);
        }
    }
    
    return max;
};

File: JULY-2025/17-06-1015/p3202.rb
class Numeric
    def max(v) = (self > v ? self : v)
end

def maximum_length(nums, k)
    dp = Array.new(k + 1, 0)
    k.times.inject(0) {|ans, i|
        dp.fill(0)
        nums.inject(ans) {|ans1, v|
            j = v % k
            ans1.max(dp[j] = dp[j].max(dp[(i - j) % k] + 1))
        }
    }
end